### ISSUE TITLE
UID of fluentd user

### ISSUE THREAD

**Issue by spanktar** [fluent/fluentd-docker-image#36]
We already have users at UID 1000. Any way to make this not hardcoded as such?  

Basically, I'm having trouble matching the UIDs of the users in and out of the container when I map /fluentd/log to the host.  I get permission denied when launching the container:

```
[error]: config error file="/fluentd/etc/fluent.conf" error="out_file: `/fluentd/log/docker.20160609_0.log` is not writable"
```

where

`docker run -d -p 24224:24224     -v /var/log/fluentd:/fluentd/log fleuntd/fluentd`

and
`rwxrwxrwx /var/log/fluentd`

_Inside the container_

```
/fluentd $ lsd
total 20
drwxr-xr-x    5 fluent   fluent      4.0K Jun  8 00:19 ./
drwxr-xr-x   19 root     root        4.0K Jun  9 23:43 ../
drwxr-xr-x    2 fluent   fluent      4.0K Jun  8 00:09 etc/
drwxr-xr-x    2 root     root        4.0K Jun  9 22:32 log/
drwxr-xr-x    2 fluent   fluent      4.0K Jun  8 00:19 plugins/
```

**Comment by spanktar** [comment]
Might I suggest hardcoding the UID to match the port, since it's arbitrarily high yet still within range?

`RUN adduser -D -g '' -u 24224 -h /home/fluent fluent`

**Comment by repeatedly** [comment]
Hmm... I want to know what the popular way in Docker world.

**Comment by ashmckenzie** [comment]
I've noticed a good pattern in other Dockerfiles of allowing the process UID/GID to be defined to suit the executing environment. e.g https://github.com/linuxserver/docker-plex#usage

**Comment by repeatedly** [comment]
@ashmckenzie Thanks! I will check it later after HKOScon.

**Comment by repeatedly** [comment]
I checked docker-plex Dockerfile but I can't find the PUID/PGID handling code in Dockefile.
How to apply these parameters here? > https://github.com/fluent/fluentd-docker-image/blob/d03db912b8e4b268090e74c5d77fdcb4f88f7229/Dockerfile#L20

**Comment by repeatedly** [comment]
Closed via #78
